.github/workflows/aws-deploy.yaml

```diff
@@ -0,0 +1,54 @@
+name: aws-deploy
+
+on:
+  pull_request:
+    branches: '*'
+  push:
+    branches: [prod]
+
+jobs:
+
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - uses: pre-commit/action@v2.0.0
+
+  sceptre-deploy:
+    if: github.event_name == 'push'
+    runs-on: ubuntu-latest
+    needs: pre-commit
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+      - name: Cache Pipenv virtualenv
```

Comment:
hmm, wouldn't it be better to decouple this cache from this job?  maybe create the cache in separate job and have downstream jobs depend on it?

Reply:
I think the Sceptre GitHub action eliminates the need for the caching step since the action uses a Docker container with Sceptre preinstalled.